app/src/contexts/transactionLogs.tsx

```diff
@@ -43,11 +43,11 @@ export function TransactionsProvider(props: { children: any }) {
 
   // Returns the correct action type string
   // For staking from airdrop or wallet
-  const getActionType = (action: string, wallet: string): string => {
+  const getActionType = (action: string): string => {
     let actionType;
     switch (action) {
       case "stake":
-        actionType = `Staked from wallet ${shortenAddress(wallet, 4)}`;
+        actionType = `Staked from wallet ${shortenAddress(publicKey!.toString(), 4)}`;
```

Comment:
Better to just have a fallback instead of letting the app crash if the wallet disconnects for whatever reason
```suggestion
        actionType = `Staked from wallet ${publicKey ? shortenAddress(publicKey.toString(), 4) : ""}`;
```